packages/react/src/components/OverflowMenu/OverflowMenu.tsx

```diff
@@ -584,9 +718,12 @@ class OverflowMenu extends Component {
   }
 }
 
-export { OverflowMenu };
-export default (() => {
-  const forwardRef = (props, ref) => <OverflowMenu {...props} innerRef={ref} />;
-  forwardRef.displayName = 'OverflowMenu';
-  return React.forwardRef(forwardRef);
-})();
+const WithFwdRef = React.forwardRef<HTMLElement, OverflowMenuProps>(
+  (props, ref) => {
+    return <OverflowMenuComponent {...props} innerRef={ref} />;
+  }
+);
+WithFwdRef.displayName = 'OverflowMenu';
+
+export const OverflowMenu = WithFwdRef;
+export default WithFwdRef;
```

Comment:
I'd prefer to not rename the component from what it is now. It's possible consumers are relying on the existing name in tests or otherwise, even with the `displayName` configured the same. Would something like this work instead?

```suggestion
export { OverflowMenu };
export default (() => {
  const forwardRef = (props, ref) => <OverflowMenu {...props} innerRef={ref} />;
  forwardRef.displayName = 'OverflowMenu';
  return React.forwardRef<HTMLElement, OverflowMenuProps>(forwardRef);
})();
```

Reply:
What is the logic behind not having the ref forwarding on named export, though? I don't really see the use-case, but I'll be the first to admit that my scope is pretty limited …

Also, why the extra function wrapper?

Reply:
There probably is not be a good reason for it honestly. It's unlikely, but someone could be relying on the named export via importing the filepath. Changing it to use forwardRef could break that usage.

```js
@import { OverflowMenu } from '@carbon/react/es/components/OverflowMenu/OverflowMenu';
```

I assume the extra function wrapper was in there simply so the forwardRef component could be given the `OverflowMenu` displayName.